### ISSUE TITLE
Review Kafka attributes 

### ISSUE THREAD

**Issue by lmolkova** [open-telemetry/semantic-conventions#751]
Based on the messaging SIG discussion on 2/15/24, we should target stabilizing Kafka semconv along with the messaging semconv stability.

While we can always add span attributes, but adding attributes to the metrics is breaking #722.

So prior to stability, we should 
- review if some important (for metrics) attributes are missing 
- have kafka metrics instrumentation prototypes.

**Comment by lmolkova** [comment]
adding research on RabbitMQ 

TL;DR:
pre-stability:
- we should consider adding `redelivered` and `delivery_mode` attributes  for RabbitMQ.
- we should consider overriding span names for rabbitmq to follow rabbitmq terminology when it does not align (`consume`, `get`, etc)
can be added incrementally:
- we should  consider adding `messaging.header.*` template similar to http headers.
- we should consider adding general-purpose attribute to record time-in-queue on spans (and generic metric). It's extremely useful and  very common.

Java otel instr reports
- [`rabbitmq.record.queue_time_ms`](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/7a4f121f8f6849c83b44ccbc7284387233314656/instrumentation/rabbitmq-2.7/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rabbitmq/RabbitDeliveryExperimentalAttributesExtractor.java#L19) - time in queue for message
- [`rabbitmq.command`](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/7a4f121f8f6849c83b44ccbc7284387233314656/instrumentation/rabbitmq-2.7/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rabbitmq/RabbitInstrumenterHelper.java#L42) (`basic.publish`, `basic.get`, `queue.create`, `exchange.declare` etc) - perhaps should be changed to `messaging.operation`
[`rabbitmq.queue`](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/7a4f121f8f6849c83b44ccbc7284387233314656/instrumentation/rabbitmq-2.7/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rabbitmq/RabbitReceiveExperimentalAttributesExtractor.java#L19) - perhaps should be changed to `messaging.destination.name`
- [`rabbitmq.delivery_mode`](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/7a4f121f8f6849c83b44ccbc7284387233314656/instrumentation/rabbitmq-2.7/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rabbitmq/RabbitInstrumenterHelper.java) - https://www.rabbitmq.com/consumers.html#message-properties
- [`messaging.header.*` template](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/778fc3eb89000f8900f1591ebbc5987056bbacb6/instrumentation-api-incubator/src/main/java/io/opentelemetry/instrumentation/api/incubator/semconv/messaging/CapturedMessageHeadersUtil.java#L33-L34)


Python otel doesn't report anything new

RabbitMQ message properties that could be interesting on metrics (low cardinality, logical level): 
- `Redelivered` bool flag is message was delivered before
- `Delivery mode` (persistent vs transient message)
- `type` - Application-specific message type, e.g. "orders.created"
- `Content type` and `Content encoding` - e.g. "application/json" and "gzip". Used by applications, not core RabbitMQ


Collector [rabbitmq scraper](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/7c5999f44f6ff7dab101cc0b3e8f167d96af20de/receiver/rabbitmqreceiver/documentation.md)  does not report any attributes.

[Client JMX metrics](https://blog.rabbitmq.com/posts/2016/11/metrics-support-in-rabbitmq-java-client-4-0/#with-jmx) don't seem to have any attributes

**Comment by pyohannes** [comment]
> we should also check rabbitmq/kafka attributes to see if any of the existing ones are good candidates to be generalized

Based on discussions in #798, we should also look into standardizing `messaging.kafka.message.offset`.

**Comment by pyohannes** [comment]
Discussed in the messaging workgroup: we'll need a RabbitMQ-specific extension for metrics because the routing key is needed on RabbitMQ metrics.

**Comment by pyohannes** [comment]
This issue is related: https://github.com/open-telemetry/semantic-conventions/issues/1156

**Comment by lmolkova** [comment]
- delivery mode - not necessary for stability, it's a constant, won't appear on metrics and we can always add it later
- redelivered - should appear on metrics (at least consumed messages). Proposal - `messaging.message.redelivered` (bool) - low cardinality, supported in many systems

> type - Application-specific message type, e.g. "orders.created"
Content type and Content encoding - e.g. "application/json" and "gzip". Used by applications, not core RabbitMQ

`messaging.message.schema.id|format|type` - appears in JMS Rabbit, Kafka Avro level, IBM MQ
